src/utils/advance.js

```diff
@@ -0,0 +1,42 @@
+import { Random } from "@woowacourse/mission-utils";
+import { GAME_RULES } from "../constants/index.js";
+
+// 전진 조건 체크
+export const shouldAdvance = (randomNumber) => {
+  return randomNumber >= GAME_RULES.ADVANCE_NUMBER;
+};
+
+// 자동차 전진 로직(전진 카운트 증가)
+export const advanceCars = (carNames, incrementAdvanceCount) => {
+  const { MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER } = GAME_RULES;
+  carNames.forEach((carName) => {
+    const randomNumber = Random.pickNumberInRange(MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER);
+
+    if (shouldAdvance(randomNumber)) {
+      incrementAdvanceCount(carName);
+    }
+  });
+};
```

Comment:
함수형으로 작성하셨다고 해서 후다닥 달려왔습니다! 이 부분에서 모듈화가 잘 이루어져있다고 생각합니다. 살짝 아쉬운 부분은 getRandomNumber를 외부에서 주입받아서 더 순수성을 높일 수 있지 않을까 하는 점입니다!

```suggestion
export const advanceCars = (carNames, getRandomNumber) => {
  return carNames.reduce((acc, carName) => {
    const randomNumber = getRandomNumber();
    if (shouldAdvance(randomNumber)) {
      return { ...acc, [carName]: (acc[carName] || 0) + 1 };
    }
    return acc;
  }, {});
};
```

이렇게 수정해 볼 수 있지 않을까요?

Reply:
`incrementAdvanceCount`를 사용하는게 핵심인 로직입니다!

`incrementAdvanceCount` 함수만으로 자동차의 정보를 수정할 수 있기 때문에 원본 객체에 대한 불변성을 지킬 수 있어서 클로저를 적극 활용했던 것 같아요🙃